### ISSUE TITLE
Why is ground-truth label all zero in irl_cost calculation?

### ISSUE THREAD

**Issue by tjx1233** [MCZhi/DTPP#13]
in train_util.py:
```
def calc_loss(neighbors, ego, ego_regularization, scores, weights, ego_gt, neighbors_gt, neighbors_valid):
    mask = torch.ne(ego.sum(-1), 0)
    neighbors = neighbors[:, 0] * neighbors_valid 
    cmp_loss = F.smooth_l1_loss(neighbors, neighbors_gt, reduction='none')
    cmp_loss = cmp_loss * mask[:, 0, None, :, None]
    cmp_loss = cmp_loss.sum() / mask[:, 0].sum()

    regularization_loss = F.smooth_l1_loss(ego_regularization, ego_gt, reduction='none')
    regularization_loss = regularization_loss * mask[:, 0, :, None]
    regularization_loss = regularization_loss.sum() / mask[:, 0].sum()

    label = torch.zeros(scores.shape[0], dtype=torch.long).to(scores.device)    
    irl_loss = F.cross_entropy(scores, label)

    weights_regularization = torch.square(weights).mean()

    loss = cmp_loss + irl_loss + 0.1 * regularization_loss + 0.01 * weights_regularization

    return loss
```
When calculating the irl_loss, the ground-truth label is set to all zero, why? Shouldn't the label come from datasets ego trajectory?

**Comment by SupOne11** [comment]
I  have the same question with you, I think the predict ego trajectory which is closest to gt should be set to 1.

**Comment by SupOne11** [comment]
> See #5. We did not include the gt ego trajectory in the tree generation, so the gt label is the generated trajectory closet to gt.
Thank you for your reply. But I still can't understand that all of the elements in label is set to zero，how to identify which ego trajectory is closet to gt

**Comment by MCZhi** [comment]
The elements in the ego trajectory tree have already been sorted in data processing, so the one closest to gt is at index 0.

**Comment by SupOne11** [comment]
> The elements in the ego trajectory tree have already been sorted in data processing, so the one closest to gt is at index 0.

So, shouldn't label[0] be set to 1, with all other elements set to 0?

**Comment by MCZhi** [comment]
The label is actually the target index, which means for all data in the batch, the ground-truth index (label) is 0.

**Comment by SupOne11** [comment]
> The label is actually the target index, which means for all data in the batch, the ground-truth index (label) is 0.

Thank you for your reply. I understand.

**Comment by tjx1233** [comment]
Thank you for reply!
